proto/babylon/btcstkconsumer/v1/query.proto

```diff
@@ -0,0 +1,155 @@
+    option (google.api.http).get = "/babylon/btcstkconsumer/v1/finality_providers/{consumer_id}";
+  }
+
+  // FinalityProvider info about one finality provider
+  rpc FinalityProvider(QueryFinalityProviderRequest) returns (QueryFinalityProviderResponse) {
+    option (google.api.http).get = "/babylon/btcstkconsumer/v1/finality_provider/{consumer_id}/{fp_btc_pk_hex}";
+  }
+}
+
+// QueryParamsRequest is request type for the Query/Params RPC method.
+message QueryParamsRequest {}
+
+// QueryParamsResponse is response type for the Query/Params RPC method.
+message QueryParamsResponse {
+  // params holds all the parameters of this module.
+  Params params = 1 [
+    (gogoproto.nullable) = false,
+    (amino.dont_omitempty) = true
+  ];
+}
+
+// QueryConsumerRegistryListRequest is request type for the Query/ConsumerRegistryList RPC method
+message QueryConsumerRegistryListRequest {
+  // pagination defines whether to have the pagination in the request
+  cosmos.base.query.v1beta1.PageRequest pagination = 1;
+}
+
+// QueryConsumerRegistryListResponse is response type for the Query/ConsumerRegistryList RPC method
+message QueryConsumerRegistryListResponse {
+  // consumer_ids are IDs of the consumers in ascending alphabetical order
+  repeated string consumer_ids = 1;
+  // pagination defines the pagination in the response
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// QueryConsumersRegistryRequest is request type for the Query/ConsumersRegistry RPC method.
+message QueryConsumersRegistryRequest { repeated string consumer_ids = 1; }
+
+// QueryConsumersRegistryResponse is response type for the Query/ConsumersRegistry RPC method.
+message QueryConsumersRegistryResponse {
+  repeated babylon.btcstkconsumer.v1.ConsumerRegister consumers_register = 1;
+}
+
+// QueryFinalityProvidersRequest is the request type for the
+// Query/FinalityProviders RPC method.
+message QueryFinalityProvidersRequest {
+  string consumer_id = 1;
+  // pagination defines an optional pagination for the request.
+  cosmos.base.query.v1beta1.PageRequest pagination = 2;
+}
+
+// QueryFinalityProvidersResponse is the response type for the
+// Query/FinalityProviders RPC method.
+message QueryFinalityProvidersResponse {
+  // finality_providers contains all the finality providers
+  repeated FinalityProviderResponse finality_providers = 1;
+
+  // pagination defines the pagination in the response.
+  cosmos.base.query.v1beta1.PageResponse pagination = 2;
+}
+
+// QueryFinalityProviderRequest requests information about a finality provider
+message QueryFinalityProviderRequest {
+  // consumer id is the consumer id this finality provider is registered to
+  string consumer_id = 1;
+  // fp_btc_pk_hex is the hex str of Bitcoin secp256k1 PK of the finality provider
+  string fp_btc_pk_hex = 2;
+}
+
+// QueryFinalityProviderResponse contains information about a finality provider
+message QueryFinalityProviderResponse {
```

Comment:
why don't we use the finality provider from `x/btcstaking`

Reply:
We created this endpoint for consumer FPs explicitly, but yes merging two is surely possible